### ISSUE TITLE
预测结果全部重复

### ISSUE THREAD

**Issue by shisg1002** [AaronJny/lotto#3]
大神， 我用最新的大乐透结果进行预测。先运行python3 train_with_whole_dataset.py，然后运行predict.py，结果预测结果全是最后一期的开奖号码（第1注 8 19 29 34 35 6 11，这个结果是更新下来的开奖数据中的最新一期结果）。这个问题出在哪？

[root@localhost lotto-master]# python3 predict.py 
2020-12-04 09:55:42.890432: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2020-12-04 09:55:42.901135: I tensorflow/core/platform/profile_utils/cpu_utils.cc:104] CPU Frequency: 1696070000 Hz
2020-12-04 09:55:42.902469: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x3da0480 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-12-04 09:55:42.902529: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
Model: "functional_1"
__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
x1 (InputLayer)                 [(None, 256, 35)]    0                                            
__________________________________________________________________________________________________
x2 (InputLayer)                 [(None, 256, 35)]    0                                            
__________________________________________________________________________________________________
x3 (InputLayer)                 [(None, 256, 35)]    0                                            
__________________________________________________________________________________________________
x4 (InputLayer)                 [(None, 256, 35)]    0                                            
__________________________________________________________________________________________________
x5 (InputLayer)                 [(None, 256, 35)]    0                                            
__________________________________________________________________________________________________
x6 (InputLayer)                 [(None, 256, 12)]    0                                            
__________________________________________________________________________________________________
x7 (InputLayer)                 [(None, 256, 12)]    0                                            
__________________________________________________________________________________________________
bidirectional (Bidirectional)   (None, 256, 128)     51200       x1[0][0]                         
__________________________________________________________________________________________________
bidirectional_2 (Bidirectional) (None, 256, 128)     51200       x2[0][0]                         
__________________________________________________________________________________________________
bidirectional_4 (Bidirectional) (None, 256, 128)     51200       x3[0][0]                         
__________________________________________________________________________________________________
bidirectional_6 (Bidirectional) (None, 256, 128)     51200       x4[0][0]                         
__________________________________________________________________________________________________
bidirectional_8 (Bidirectional) (None, 256, 128)     51200       x5[0][0]                         
__________________________________________________________________________________________________
bidirectional_10 (Bidirectional (None, 256, 128)     39424       x6[0][0]                         
__________________________________________________________________________________________________
bidirectional_12 (Bidirectional (None, 256, 128)     39424       x7[0][0]                         
__________________________________________________________________________________________________
dropout (Dropout)               (None, 256, 128)     0           bidirectional[0][0]              
__________________________________________________________________________________________________
dropout_2 (Dropout)             (None, 256, 128)     0           bidirectional_2[0][0]            
__________________________________________________________________________________________________
dropout_4 (Dropout)             (None, 256, 128)     0           bidirectional_4[0][0]            
__________________________________________________________________________________________________
dropout_6 (Dropout)             (None, 256, 128)     0           bidirectional_6[0][0]            
__________________________________________________________________________________________________
dropout_8 (Dropout)             (None, 256, 128)     0           bidirectional_8[0][0]            
__________________________________________________________________________________________________
dropout_10 (Dropout)            (None, 256, 128)     0           bidirectional_10[0][0]           
__________________________________________________________________________________________________
dropout_12 (Dropout)            (None, 256, 128)     0           bidirectional_12[0][0]           
__________________________________________________________________________________________________
bidirectional_1 (Bidirectional) (None, 256, 128)     98816       dropout[0][0]                    
__________________________________________________________________________________________________
bidirectional_3 (Bidirectional) (None, 256, 128)     98816       dropout_2[0][0]                  
__________________________________________________________________________________________________
bidirectional_5 (Bidirectional) (None, 256, 128)     98816       dropout_4[0][0]                  
__________________________________________________________________________________________________
bidirectional_7 (Bidirectional) (None, 256, 128)     98816       dropout_6[0][0]                  
__________________________________________________________________________________________________
bidirectional_9 (Bidirectional) (None, 256, 128)     98816       dropout_8[0][0]                  
__________________________________________________________________________________________________
bidirectional_11 (Bidirectional (None, 256, 128)     98816       dropout_10[0][0]                 
__________________________________________________________________________________________________
bidirectional_13 (Bidirectional (None, 256, 128)     98816       dropout_12[0][0]                 
__________________________________________________________________________________________________
dropout_1 (Dropout)             (None, 256, 128)     0           bidirectional_1[0][0]            
__________________________________________________________________________________________________
dropout_3 (Dropout)             (None, 256, 128)     0           bidirectional_3[0][0]            
__________________________________________________________________________________________________
dropout_5 (Dropout)             (None, 256, 128)     0           bidirectional_5[0][0]            
__________________________________________________________________________________________________
dropout_7 (Dropout)             (None, 256, 128)     0           bidirectional_7[0][0]            
__________________________________________________________________________________________________
dropout_9 (Dropout)             (None, 256, 128)     0           bidirectional_9[0][0]            
__________________________________________________________________________________________________
dropout_11 (Dropout)            (None, 256, 128)     0           bidirectional_11[0][0]           
__________________________________________________________________________________________________
dropout_13 (Dropout)            (None, 256, 128)     0           bidirectional_13[0][0]           
__________________________________________________________________________________________________
time_distributed (TimeDistribut (None, 256, 105)     13545       dropout_1[0][0]                  
__________________________________________________________________________________________________
time_distributed_1 (TimeDistrib (None, 256, 105)     13545       dropout_3[0][0]                  
__________________________________________________________________________________________________
time_distributed_2 (TimeDistrib (None, 256, 105)     13545       dropout_5[0][0]                  
__________________________________________________________________________________________________
time_distributed_3 (TimeDistrib (None, 256, 105)     13545       dropout_7[0][0]                  
__________________________________________________________________________________________________
time_distributed_4 (TimeDistrib (None, 256, 105)     13545       dropout_9[0][0]                  
__________________________________________________________________________________________________
time_distributed_5 (TimeDistrib (None, 256, 36)      4644        dropout_11[0][0]                 
__________________________________________________________________________________________________
time_distributed_6 (TimeDistrib (None, 256, 36)      4644        dropout_13[0][0]                 
__________________________________________________________________________________________________
flatten (Flatten)               (None, 26880)        0           time_distributed[0][0]           
__________________________________________________________________________________________________
flatten_1 (Flatten)             (None, 26880)        0           time_distributed_1[0][0]         
__________________________________________________________________________________________________
flatten_2 (Flatten)             (None, 26880)        0           time_distributed_2[0][0]         
__________________________________________________________________________________________________
flatten_3 (Flatten)             (None, 26880)        0           time_distributed_3[0][0]         
__________________________________________________________________________________________________
flatten_4 (Flatten)             (None, 26880)        0           time_distributed_4[0][0]         
__________________________________________________________________________________________________
flatten_5 (Flatten)             (None, 9216)         0           time_distributed_5[0][0]         
__________________________________________________________________________________________________
flatten_6 (Flatten)             (None, 9216)         0           time_distributed_6[0][0]         
__________________________________________________________________________________________________
dense_1 (Dense)                 (None, 105)          2822505     flatten[0][0]                    
__________________________________________________________________________________________________
dense_3 (Dense)                 (None, 105)          2822505     flatten_1[0][0]                  
__________________________________________________________________________________________________
dense_5 (Dense)                 (None, 105)          2822505     flatten_2[0][0]                  
__________________________________________________________________________________________________
dense_7 (Dense)                 (None, 105)          2822505     flatten_3[0][0]                  
__________________________________________________________________________________________________
dense_9 (Dense)                 (None, 105)          2822505     flatten_4[0][0]                  
__________________________________________________________________________________________________
dense_11 (Dense)                (None, 36)           331812      flatten_5[0][0]                  
__________________________________________________________________________________________________
dense_13 (Dense)                (None, 36)           331812      flatten_6[0][0]                  
__________________________________________________________________________________________________
concatenate (Concatenate)       (None, 525)          0           dense_1[0][0]                    
                                                                 dense_3[0][0]                    
                                                                 dense_5[0][0]                    
                                                                 dense_7[0][0]                    
                                                                 dense_9[0][0]                    
__________________________________________________________________________________________________
concatenate_1 (Concatenate)     (None, 72)           0           dense_11[0][0]                   
                                                                 dense_13[0][0]                   
__________________________________________________________________________________________________
y1 (Dense)                      (None, 35)           18410       concatenate[0][0]                
__________________________________________________________________________________________________
y2 (Dense)                      (None, 35)           18410       concatenate[0][0]                
__________________________________________________________________________________________________
y3 (Dense)                      (None, 35)           18410       concatenate[0][0]                
__________________________________________________________________________________________________
y4 (Dense)                      (None, 35)           18410       concatenate[0][0]                
__________________________________________________________________________________________________
y5 (Dense)                      (None, 35)           18410       concatenate[0][0]                
__________________________________________________________________________________________________
y6 (Dense)                      (None, 12)           876         concatenate_1[0][0]              
__________________________________________________________________________________________________
y7 (Dense)                      (None, 12)           876         concatenate_1[0][0]              
==================================================================================================
Total params: 15,973,524
Trainable params: 15,973,524
Non-trainable params: 0
__________________________________________________________________________________________________
本次预测结果如下：
第1注 8 19 29 34 35 6 11
第2注 8 19 29 34 35 6 11
第3注 8 19 29 34 35 6 11
第4注 8 19 29 34 35 6 11
第5注 8 19 29 34 35 6 11
第6注 8 19 29 34 35 6 11
第7注 8 19 29 34 35 6 11
第8注 8 19 29 34 35 6 11
第9注 8 19 29 34 35 6 11
第10注 8 19 29 34 35 6 11

**Comment by zeku2022** [comment]
训练的不够吧?

**Comment by fei15115** [comment]
我在我的环境下训练了400次。出来的结果也非常相似。
第1注 7 10 12 14 32 4 11
第2注 7 10 12 14 32 4 11
第3注 7 10 14 19 32 4 11
第4注 7 10 14 19 32 4 11
第5注 7 10 12 14 30 4 11

**Comment by zeku2022** [comment]
经过我多次实践，结果好像都是这个数。。。。 8 19 29 34 35 6 11类似
篮球预测都是一样的  6  11
